WangscapeTest/TestPowIntSigned.cpp

```diff
@@ -0,0 +1,102 @@
+#include <gtest/gtest.h>
+
+#include <type_traits>
+
+#include <utils.h>
+
+template<typename I>
+class TestPowIntSigned : public ::testing::Test
+{
+protected:
+    const static I min = std::numeric_limits<I>::min();
+    const static I minus_five = -5;
+    const static I minus_three = -3;
+    const static I minus_two = -2;
+    const static I minus_one = -1;
+    const static I zero = 0;
+    const static I one = 1;
+    const static I two = 2;
+    const static I three = 3;
+    const static I five = 5;
+    const static I digits = std::numeric_limits<I>::digits;
+    const static I max = std::numeric_limits<I>::max();
```

Comment:
The `TYPED_TEST` macros access `I` as `TypeParam`. I would prefer not to have all the `IntPow` tests make calls like `EXPECT_EQ((TypeParam)1, pow((TypeParam)0, (TypeParam)0));`, nor have to put `typedef TypeParam I;` at the start of every test case. `this->zero` is not ideal, but I prefer it to the alternatives. 

I could try giving `TestPowInt<I>` a member method `I callPow(I base, I exp) {return pow<I>(base, exp);}`. Then the test cases could do something like `EXPECT_EQ((TypeParam)1, this->callPow(0, 0));`. That would still have one `TypeParam` per line, though.

I can reduce the number of checks made, but they're fast enough not to have any cost associated. `base == 2` and `base == -2` are treated differently in `pow`, so `3` and `-3` will still be needed in some tests. `5` can probably be removed. `digits` and `max` are limits, although some of the combinations they're used in can be removed.